pypeit/core/save.py

```diff
@@ -351,9 +134,58 @@ def save_1d_spectra_hdf5(slf, fitsdict, clobber=True):
     # Dump into a linetools.spectra.xspectrum1d.XSpectrum1D
 '''
 
-def save_1d_spectra_fits(specObjs, header, pypeline, instrume,
-                         outfile, helio_dict=None, telescope=None, overwrite=True,
-                         update_det=None):
+
+def save_all(sci_dict, master_key_dict, master_dir, spectrograph, head1d, head2d, scipath, basename,
+                  only_1d=False, refframe='heliocentric', update_det=None, binning=None):
+
+    # Filenames to write out
+    objinfofile = scipath + '/objinfo_{:s}.txt'.format(basename)
+    outfile1d = os.path.join(scipath, 'spec1d_{:s}.fits'.format(basename))
+    outfile2d = scipath + '/spec2d_{:s}.fits'.format(basename)
+
+    # TODO: Need some checks here that the exposure has been reduced
+
+    # Build the final list of specobjs and vel_corr
+    all_specobjs = specobjs.SpecObjs()
+
+    vel_corr = 0.  # This will not be set for Standard stars, which is fine
+    for key in sci_dict:
+        if key in ['meta']:
+            vel_corr = sci_dict['meta']['vel_corr']
+            continue
+        #
+        try:
+            all_specobjs.add_sobj(sci_dict[key]['specobjs'])
+        except KeyError:  # No object extracted
+            continue
+
+    if len(all_specobjs) == 0:
+        msgs.warn('No objects to save!')
+        return
+
+    # Create the helio_dict
+    helio_dict = dict(refframe=refframe, vel_correction=vel_corr)
+
+    save_1d_spectra_fits(all_specobjs, head1d, spectrograph, outfile1d,helio_dict=helio_dict, update_det=update_det)
+
+    # 1D only?
+    if only_1d:
+        return
+    # Obj info
+    save_obj_info(all_specobjs, spectrograph, objinfofile, binning=binning)
+    # Write 2D images for the Science Frame
+
+    # TODO: Make sure self.det is correct!
+    # master_key = self.fitstbl.master_key(frame, det=self.det)
+    # TODO: Why is the raw file header needed?  Can the data be gotten from fitstbl?
+    #  If not, is it worth adding the relevant information to fitstbl?
+    # Original header
+    save_2d_images(sci_dict, head2d, master_key_dict, master_dir, outfile2d, update_det=update_det)
+
+    return
+
+
+def save_1d_spectra_fits(specObjs, header, spectrograph, outfile, helio_dict=None, overwrite=True, update_det=None):
```

Comment:
We might debate whether to pass `spectrograph` at the core level.

I think it is ok for I/O.  Not good for other core routines.

Reply:
Yes I generally completely agree, but this simplifies the syntax considerably.